### ISSUE TITLE
Big Blue Button integration doesn't work on Chrome when Oauth2 is on

### ISSUE THREAD

**Issue by taksan** [thecodingmachine/workadventure#2325]
WA version: 1.11.2

I just configured BBB and setup BBB integration. I setup BBB to authenticate with Google Oauth2. While it works flawlessly on Firefox, it does't work on Google Chrome and other chromium base browsers (for example, it doesn't work on the desktop client). I attached a screen shot.

![image](https://user-images.githubusercontent.com/96873/174341552-d97ec59e-6802-4334-8e53-ad46f97518a2.png)

Also. here's a debug console screen shot:

![image](https://user-images.githubusercontent.com/96873/174341204-dbdf3659-8d0f-4ee4-a797-e1045fc73e60.png)

**Comment by fcecagno** [comment]
Hey, I don't think it has to do with Google OAuth2, but with the cookie policy.
BigBlueButton uses a cookie to authenticate the user, and this cookie might be blocked by your browser IF your BigBlueButton server is hosted in a domain different than the iframe parent (your WA installation).
You might try to flip this config in your BigBlueButton server to `true` in order to confirm: https://github.com/bigbluebutton/bigbluebutton/blob/v2.5.1/bigbluebutton-web/grails-app/conf/bigbluebutton.properties#L311=

**Comment by fcecagno** [comment]
Hi @taksan , just to confirm, are you using the default BigBlueButton server (test-install)? I just recall it had some sort of limitation when embed on an iframe.

**Comment by taksan** [comment]
Thanks for the quick reply @fcecagno 

I was using an instance I installed by myself. I just wanted to compare BBB and Jitsi. I'm facing other issues where some users cannot see the shared screen and the sound quality is pretty bad, loud and broken.

**Comment by KlausKnopper** [comment]
> Hey, I don't think it has to do with Google OAuth2, but with the cookie policy. BigBlueButton uses a cookie to authenticate the user, and this cookie might be blocked by your browser IF your BigBlueButton server is hosted in a domain different than the iframe parent (your WA installation). You might try to flip this config in your BigBlueButton server to `true` in order to confirm: https://github.com/bigbluebutton/bigbluebutton/blob/v2.5.1/bigbluebutton-web/grails-app/conf/bigbluebutton.properties#L311=

Unfortunately this workaround is only possible for people who run their own instance of BBB and have admin access. Also, the proposed config change in BBB opens security issues.

Is there a workaround possible for setting the BBB session cookie explicitly from a self-hosted script, to make BBB sessions work in Workadventure? Or, maybe open a browser tab (where cookie domains should work) instead of a frame?

**Comment by fcecagno** [comment]
> Unfortunately this workaround is only possible for people who run their own instance of BBB and have admin access. Also, the proposed config change in BBB opens security issues.

You're right, it was just a suggestion to test, but shouldn't be considered an option for real world use.

In order to set all BigBlueButton cookies with policy `SameSite=None`, you can try this:
- patch `/usr/share/bigbluebutton/nginx/web`:
```
***************
*** 1,6 ****
--- 1,7 ----
        # Handle request to bbb-web running within a SpringBoot Tomcat embedded servlet container.  This is for BBB-API and Presentation.
        location /bigbluebutton {
                proxy_http_version 1.1;
+               proxy_cookie_path / "/; secure; SameSite=None";
  
                location /bigbluebutton {
                        proxy_pass         http://127.0.0.1:8090;
```
- reload nginx with `sudo systemctl reload nginx`.

It should do the trick.

I'd suggest to change the issue title to `BigBlueButton doesn't load on Chrome`, so it can benefit others with the same issue.

**Comment by fcecagno** [comment]
@taksan did it do the trick?

**Comment by taksan** [comment]
Will test this weekend
